[DOCS] Release highlight for security-on-by-default (#83292)
elastic/elasticsearch: docs/reference/release-notes/highlights.asciidoc

// should be featured in the Stack Installation and Upgrade Guide:

// tag::notable-highlights[] 
[discrete]
=== Security features are enabled and configured by default

Running {es} without security leaves your cluster exposed to anyone who can send
network traffic to {es}. In previous versions, you had to explicitly enable the 
{es} security features such as authentication, authorization, and network
encryption (TLS). Starting in {es} 8.0, security is
{ref}/configuring-stack-security.html[enabled and configured by default] when
you start {es} for the first time. 

At startup, we generate enrollment tokens so that you can connect a {kib} 
instance or enroll additional nodes in your secured {es} cluster, without having 
to generate security certificates or update `yml` configuration files. Just use 
the generated enrollment token when starting new nodes or {kib} instances, and 
the {stack} handles all of the security configuration for you. Out of the box, 
you'll get authentication, authorization, and network encryption with TLS for 

> internode communications between {es} and {kib}. You can create new enrollment 

Reviewer: ```suggestion
internode communications and communications between {es} and {kib}. You can create new enrollment 
```

not sure if there is a more concise way to say the above but "internode" == between es nodes , which is not the same thing as between es and kibana

Author: I decided to change this sentence to a bulleted list and provide more detail. I think you're right that we need to call out that you get encrypted communications both between nodes and between Elasticsearch and Kibana, so the bulleted list seemed like a better option. Plus, it's easier to scan!

// should be featured in the Stack Installation and Upgrade Guide:

// tag::notable-highlights[] 
[discrete]

> === Security is enabled and configured by default

Reviewer: ```suggestion
=== Security features are enabled and configured by default
```

Running {es} without security leaves your cluster exposed to anyone who can send

> network traffic to {es}. In previous versions, you had to enable the {es}
> security features and then configure password protection and Transport Layer
> Security (TLS).

Reviewer: ```suggestion
network traffic to {es}. In previous versions, you had to explicitly enable the {es}
security features such as authantication, authorization and network encryption (TLS).
```

Starting in {es} 8.0, security is
{ref}/configuring-stack-security.html[enabled and configured by default] when

> you start {es} for the first time. We also generate an enrollment token at

Reviewer: I think it would be clearer if we don't make the distinction between the tokens in such a way (For instance, we generate an enrollment token for nodes in docker, but reading this makes you think that you have to do something additional in that case too ). How about we say something more generic similar to:

"We generate enrollment tokens so that you can quickly and easily connect a {kib} instance 
or additional nodes to your secured {es} cluster without having to generate security certificates or update
`yml` configuration files. 
Just use the enrollment token when starting new nodes or {kib} instances and the elastic stack handles all of the security configuration for you. Out of the box, you’ll get authentication, authorization, network encryption 
with TLS for internode communications and communications between {es} and {kib}.
You can always create new enrollment tokens for {es} nodes and {kib} instances using the {ref}/create-enrollment-token.html[`elasticsearch-create-enrollment-token`] tool."

WDYT ?

Author: I like your suggestions @jkakavas -- I'll incorporate them in my next push.

startup for {kib} so that you can quickly and easily connect a {kib} instance to your
secured {es} cluster without having to generate security certificates or update
`yml` configuration files. Just by starting {es}, you’ll have password
protection, internode communication secured with TLS, and encrypted connections
between {es} and {kib}.

We're also shipping the
{ref}/create-enrollment-token.html[`elasticsearch-create-enrollment-token`] tool,
which you can use to generate enrollment tokens to enroll new {es} nodes with an
existing cluster, or configure {kib} instances to communicate with an existing
{es} cluster that has security features enabled. Just pass the enrollment token
when starting new nodes and {es} handles all of the security configuration for

> you.

Reviewer 2: Do we need this whole paragraph at all? This flow is not a primary one for the getting started experience, and it's documented in the linked guide. I'd rather keep this entry short and point people to read more if they need.

Author: I'm ok with removing this paragraph. I was curious if we wanted to highlight how easy it is to enroll additional nodes, but, let's remove.
